pom.xml

```diff
@@ -1605,6 +1605,186 @@
       </build>
     </profile>
 
+    <profile>
+      <id>jacoco-check</id>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.jacoco</groupId>
+            <artifactId>jacoco-maven-plugin</artifactId>
+            <version>${maven.jacoco.plugin.version}</version>
+            <executions>
+              <execution>
+                <id>default-instrument</id>
+                <goals>
+                  <goal>instrument</goal>
+                </goals>
+                <phase>test-compile</phase>
+              </execution>
+              <execution>
+                <id>default-restore-instrumented-classes</id>
+                <goals>
+                  <goal>restore-instrumented-classes</goal>
+                </goals>
+              </execution>
+              <execution>
+                <id>default-report</id>
+                <phase>site</phase>
+                <goals>
+                  <goal>report</goal>
+                </goals>
+              </execution>
+              <execution>
+                <id>default-check</id>
+                <goals>
+                  <goal>check</goal>
+                </goals>
+                <configuration>
+                  <dataFile>${project.build.directory}/jacoco/jacoco.exec</dataFile>
+                  <rules>
+                    <rule>
+                      <element>CLASS</element>
+                      <excludes>
+                        <!-- Terence Parr, The Definitive ANTLR 4 Reference, 2.4 Building Language Applications Using Parse Trees:
+                             "To better support access to the elements within specific nodes, ANTLR generates
+                             a RuleNode subclass for each rule. ... Given this description of the concrete types, we could write
+                             code by hand to perform a depth-first walk of the tree. We could perform whatever actions we
+                             wanted as we discovered and finished nodes. Typical operations are things such as
+                             computing results, updating data structures, or generating output."
+                             We don't need these classes as we use only ParseTree methods for walking. -->
+                        <exclude>com.puppycrawl.tools.checkstyle.grammars.*</exclude>
+                        <exclude>com.puppycrawl.tools.checkstyle.grammars.javadoc.*</exclude>
```

Comment:
please enforce strict coverage for this classes, as it is done in cobertura.
We use this to control that grammar changes actually covered in most required parts. We will never have 100%, but at least this is reminder on grammar changes to cover cases with tests.

Reply:
done

Reply:
I still see wildcard exclude, please recheck that you pushed all changes

Reply:
Done, except for JavadocParser which has too many inner classes.